Updated schema doc, linked agent concept doc to schema, and minor improvements to order and wording (#2661)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/concepts/native-ai-agent-support.mdx

# AI agent support

@include 'alerts/enterprise-only.mdx'

> This feature requires Vault Enterprise 2.0.3 or later.

Reviewer: ```suggestion
```

@include 'alerts/beta.mdx'

> 

Reviewer: ```suggestion

Agent registry requires Vault Enterprise 2.0.3 or later.
```
Edited to match previous suggestion

> Agent Registry and OAuth resource server configuration require Vault Enterprise 2.0.3 or later.

Reviewer: ```suggestion
Agent Registry and OAuth resource server configurations require Vault Enterprise 2.0.3 or later.
```
Suggest moving below the beta include

# AI agent support

@include 'alerts/enterprise-only.mdx'

<Note>

This feature requires Vault Enterprise 2.0.3 or later.

> </Note>

Reviewer: ```suggestion
```
See previous note

@include 'alerts/enterprise-only.mdx'

> Agent Registry and OAuth resource server configuration require Vault Enterprise 2.0.3 or later.

Reviewer 2: "and OAuth Resource Server"

Author: Per style guide, will not capitalize. Vault features should not be uppercase

client to a Vault Identity entity, checks the entity's Agent Registry
record, and enforces authorization constraints for agentic workflows.

> These capabilities contribute their own audit metadata. See [OAuth JWT metadata](/vault/docs/audit/schema#oauth-jwt-metadata) for a specification of the detailed audit metadata. When a client authenticates with an OAuth JWT, Vault does not create a stored

Reviewer: What are "these capabilities"? We should avoid cross-paragraph pronouns

Author: Resolved. I now explicitly refer to Agent Registry and Oauth resource server

token with an operator-defined metadata map. Instead, Vault derives the token's

> metadata from the JWT's own claims each time the client presents the credential.

Reviewer: ```suggestion
token with an operator-defined metadata map. Instead, Vault derives token
metadata from claims from the JWT each time the client presents the credential.
```
Style correction: avoid possessives
